### PR TITLE
Add tls

### PR DESCRIPTION
**Pull request by R-Lawton**
Form view of the create:
Screenshot 2024-09-23 at 15 42 06

Yaml view of the create  (schema no showing trying to figure out why)
Screenshot 2024-09-23 at 15 42 19

Follow up:

- Update:  https://github.com/Kuadrant/console-plugin/issues/59
- Validation: https://github.com/Kuadrant/console-plugin/issues/58

### CODE REVIEW

File: src/components/KuadrantTLSPage.tsx [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    },
+    spec: {
+      targetRef: {
+        group: 'gateway.networking.k8s.io',
+        kind: 'Gateway',
+        name: selectedGateway,
+      },
+      issuerRef: {
+        name: selectedCertIssuer,
+        kind: 'ClusterIssuer',
```
Comment (jasonmadigan) [inline src/components/KuadrantTLSPage.tsx:82]: I think these can also use NS scoped `Issuer`'s too?

File: src/components/KuadrantTLSPage.tsx [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+      <Helmet>
+        <title data-test="example-page-title"> Create TLS Policy </title>
+      </Helmet>
+      <Page>
+        <PageSection className='pf-m-no-padding'>
+          <div className='co-m-nav-title'>
+
+            <Title headingLevel="h1">{('Create TLS Policy')}</Title>
+            <p className='help-block co-m-pane__heading-help-text'>
+              <div> Targets Gateway API networking resources Gateways to provide tls for gateway listeners by managing the lifecycle of tls certificates using CertManager</div>
```
Comment (jasonmadigan) [inline src/components/KuadrantTLSPage.tsx:179]: probably want to i18n this in the very excellently named `locales/en/plugin__console-plugin-template.json`

File: src/components/KuadrantTLSPage.tsx [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+              </NamespaceSelect>
+              <FormGroup
+                label="Gateway API Target Reference"
+                fieldId="gateway-select"
+                isRequired
+              >
+                <FormHelperText>
+                  <HelperText>
+                    <HelperTextItem>
+                      Reference to a Kubernetes resource that the policy attaches to. To create an additional gateway got to
```
Comment (jasonmadigan) [inline src/components/KuadrantTLSPage.tsx:243]: i18n

File: src/components/KuadrantTLSPage.tsx [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+              </FormGroup>
+              <FormGroup
+                label="CertIssuer Issuer Reference"
+                fieldId="certmanger-select"
+                isRequired
+              >
+                <FormHelperText>
+                  <HelperText>
+                    <HelperTextItem>
+                      Reference to the issuer for the created certificate. To create an additional Issuer got to
```
Comment (jasonmadigan) [inline src/components/KuadrantTLSPage.tsx:268]: i18n

File: src/utils/cancel.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { History } from 'history';
+
+export function handleCancel (namespace: string, policyType:string ,data , history: History)  {
+  history.push(`/k8s/all-namespaces/kuadrant.io~v1alpha1~TLSPolicy`)
```
Comment (jasonmadigan) [inline src/utils/cancel.tsx:4]:
```suggestion
  history.push(`/k8s/all-namespaces/kuadrant.io~v1alpha1~TLSPolicy`)
```

maybe something like

```suggestion
  history.push(`/k8s/all-namespaces/kuadrant.io~v1alpha1~${policyType}`)
```

guess we'd need an apiVersion too

File: src/components/KuadrantTLSPage.tsx (hunk 1 above)
Comment (R-Lawton) [reply src/components/KuadrantTLSPage.tsx:82]: do you mean you can specify a NS for the issuer?

Comment (jasonmadigan) [reply src/components/KuadrantTLSPage.tsx:82]: No i mean cert-manager can use either a `ClusterIssuer` (cluster-scoped) or an `Issuer` (ns-scoped)

Comment (R-Lawton) [reply src/components/KuadrantTLSPage.tsx:82]: oh i did not know there was two types makes sense ill change that

File: src/components/ratelimitpolicy/AddLimitModal.tsx [hunk 6]
```diff
@@ -72,7 +72,7 @@ const AddLimitModal: React.FC<{
       onClose={onClose}
       title={t('Add Limit')}
       width="50%"
-      hasNoBodyWrapper
+
```
Comment (R-Lawton) [inline src/components/ratelimitpolicy/AddLimitModal.tsx:75]: this was giving me errors

File: locales/en/plugin__console-plugin-template.json [hunk 7]
```diff
@@ -88,5 +88,16 @@
   "API Designer": "API Designer",
   "cert-manager Operator": "cert-manager Operator",
   "APIs / HTTPRoutes": "APIs / HTTPRoutes",
-  "Hide for session": "Hide for session"
+  "Hide for session": "Hide for session",
+  "Targets Gateway API networking resources Gateways to provide tls for gateway listeners by managing the lifecycle of tls certificates using CertManager": "Targets Gateway API networking resources Gateways to provide tls for gateway listeners by managing the lifecycle of tls certificates using CertManager",
```
Comment (jasonmadigan) [inline locales/en/plugin__console-plugin-template.json:92]:
```suggestion
  "Targets Gateway API networking resources Gateways to provide tls for gateway listeners by managing the lifecycle of TLS certificates using cert-manager": "Targets Gateway API networking resources Gateways to provide TLS for gateway listeners by managing the lifecycle of TLS certificates using cert-manager",
```

File: src/components/KuadrantTLSCreatePage.tsx [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+      <Helmet>
+        <title data-test="example-page-title">{t(' Create TLS Policy')} </title>
+      </Helmet>
+      <Page>
+        <PageSection className='pf-m-no-padding'>
+          <div className='co-m-nav-title'>
+
+            <Title headingLevel="h1">{t('Create TLS Policy')}</Title>
+            <p className='help-block co-m-pane__heading-help-text'>
+              <div>{t(' Targets Gateway API networking resources Gateways to provide tls for gateway listeners by managing the lifecycle of tls certificates using CertManager')}</div>
```
Comment (jasonmadigan) [inline src/components/KuadrantTLSCreatePage.tsx:160]:
```suggestion
              <div>{t(' Targets Gateway API networking resources Gateways to provide TLS for gateway listeners by managing the lifecycle of TLS certificates using cert-manager')}</div>
```

File: locales/en/plugin__console-plugin-template.json [hunk 9]
```diff
@@ -88,5 +88,16 @@
...
+  "Create TLS Policy":"Create TLS Policy",
+  "Configure via":"Configure via",
+  "Unique name of the TLSPolicy.":"Unique name of the TLSPolicy.",
+  "Reference to a Kubernetes resource that the policy attaches to. To create an additional gateway got to":"Reference to a Kubernetes resource that the policy attaches to. To create an additional gateway got to",
```
Comment (jasonmadigan) [inline locales/en/plugin__console-plugin-template.json:96]:
```suggestion
  "Reference to a Kubernetes resource that the policy attaches to. To create an additional gateway, go to":"Reference to a Kubernetes resource that the policy attaches to. To create an additional gateway, go to",
```

File: src/components/KuadrantTLSCreatePage.tsx [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+              </NamespaceSelect>
+              <FormGroup
+                label={t("Gateway API Target Reference")}
+                fieldId="gateway-select"
+                isRequired
+              >
+                <FormHelperText>
+                  <HelperText>
+                    <HelperTextItem>
+                      {t('Reference to a Kubernetes resource that the policy attaches to. To create an additional gateway got to')}
```
Comment (jasonmadigan) [inline src/components/KuadrantTLSCreatePage.tsx:224]:
```suggestion
                      {t('Reference to a Kubernetes resource that the policy attaches to. To create an additional gateway, go to')}
```

File: src/components/KuadrantTLSCreatePage.tsx [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+              </FormGroup>
+              <FormGroup
+                label={t("CertIssuer Issuer Reference")}
+                fieldId="certmanger-select"
+                isRequired
+              >
+                <FormHelperText>
+                  <HelperText>
+                    <HelperTextItem>
+                      {t('Reference to the issuer for the created certificate. To create an additional Issuer got to')}
```
Comment (jasonmadigan) [inline src/components/KuadrantTLSCreatePage.tsx:239]:
```suggestion
                      {t('Reference to the issuer for the created certificate. To create an additional Issuer, go to')}
```

File: locales/en/plugin__console-plugin-template.json [hunk 12]
```diff
@@ -88,5 +88,16 @@
...
+  "CertIssuer Issuer Reference":"CertIssuer Issuer Reference",
+  "Reference to the issuer for the created certificate. To create an additional Issuer got to":"Reference to the issuer for the created certificate. To create an additional Issuer got to",
```
Comment (jasonmadigan) [inline locales/en/plugin__console-plugin-template.json:98]:
```suggestion
  "Reference to the issuer for the created certificate. To create an additional Issuer, go to":"Reference to the issuer for the created certificate. To create an additional Issuer, go to",
```

**Review by jasonmadigan — APPROVED** [review]
couple of small typos and suggestions, otherwise looks good to merge. gave it a spin locally.

**Pull request merged**